### PR TITLE
WinNT5.1 fixes

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Please update your commit messages to follow our [guidelines for commit messages](https://github.com/OpenRCT2/OpenRCT2/wiki/Commit-Messages) :wink:

File: src/openrct2/platform/Platform.Win32.cpp [hunk 1]
```diff
@@ -37,6 +37,10 @@
 
 #    include <iterator>
 
+#    if _WIN32_WINNT < 0x600
+#        define swprintf_s(a, b, c, d, ...) swprintf(a, b, c, ##__VA_ARGS__)
```
Comment (Broxzier) [inline src/openrct2/platform/Platform.Win32.cpp:41]: This drops the fourth argument, doesn't that mean that where this is used it prints garbage now?

Comment (janisozaur) [reply src/openrct2/platform/Platform.Win32.cpp:41]: The fourth argument is buffer length, unused in the non-`_s` version.

Comment (ZehMatt) [reply src/openrct2/platform/Platform.Win32.cpp:41]: I would rather like to see this not being a macro, you can use variadic template arguments and perfect forwarding.

Comment (janisozaur) [reply src/openrct2/platform/Platform.Win32.cpp:41]: That would just shift the ifdef into a template, is that what you mean?

Comment (ZehMatt) [reply src/openrct2/platform/Platform.Win32.cpp:41]: Just the function you still need the if _WIN32_WINNT < 0x600

**Comment by AaronVanGeffen** [comment]
@janisozaur Is this ready to be merged?

**Comment by janisozaur** [comment]
@AaronVanGeffen yes. I'm still trying to figure out the problem with our CI, but that's another PR.

**Pull request merged**